src/VisualStudio/Core/Def/SampleToolWindow/SampleToolboxUserControl.xaml.cs

```diff
@@ -0,0 +1,425 @@
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    using Workspace = Microsoft.CodeAnalysis.Workspace;
+
+    /// <summary>
+    /// Interaction logic for SampleToolboxUserControl.xaml
+    /// </summary>
+    internal partial class SampleToolboxUserControl : UserControl, IOleCommandTarget
+    {
+        public SampleToolboxUserControl(Workspace workspace, IDocumentTrackingService documentTrackingService, ILanguageServiceBroker2 languageServiceBroker, IThreadingContext threadingContext)
+        {
+            InitializeComponent();
+            InitializeIfNeeded(workspace, documentTrackingService, languageServiceBroker, threadingContext);
+        }
+
+        private Workspace? workspace { get; set; }
+        private DocumentId? lastDocumentId { get; set; }
+
+        [MemberNotNullWhen(true, nameof(symbolsTreeItemsSource))]
+        private bool symbolTreeInitialized { get; set; }
+        private ObservableCollection<DocumentSymbolViewModel>? symbolsTreeItemsSource { get; set; }
+
+        private ITextSnapshot? snapshot { get; set; }
+        private IWpfTextView? textView { get; set; }
+
+        private void InitializeIfNeeded(Workspace workspace, IDocumentTrackingService documentTrackingService, ILanguageServiceBroker2 languageServiceBroker, IThreadingContext threadingContext)
+        {
+            this.workspace = workspace;
+            documentTrackingService.ActiveDocumentChanged += DocumentTrackingService_ActiveDocumentChanged;
+
+            this.textView = GetActiveTextView();
+            if (this.textView is not null)
+            {
+                this.textView.Caret.PositionChanged += FollowCursor;
+            }
+
+            searchBox.GotFocus += RemoveText;
+            searchBox.LostFocus += AddText;
+
+            void DocumentTrackingService_ActiveDocumentChanged(object sender, DocumentId? documentId)
+            {
+                threadingContext.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    this.symbolTreeInitialized = false;
+                    if (documentId == this.lastDocumentId)
+                    {
+                        return;
+                    }
+
+                    this.lastDocumentId = documentId;
+                    var document = workspace.CurrentSolution.GetDocument(documentId);
+                    if (document is null)
+                    {
+                        symbolTree.ItemsSource = new List<DocumentSymbolViewModel>();
+                        return;
+                    }
+
+                    document.TryGetText(out var text);
+                    if (text is null)
+                    {
+                        symbolTree.ItemsSource = new List<DocumentSymbolViewModel>();
+                        return;
+                    }
+
+                    var textBuffer = text.Container.GetTextBuffer();
```

Comment:
instead of doing all this to get the buffer we can just do `var textBuffer = this.textView.TextBuffer;`